### ISSUE TITLE
List of available javadocs

### ISSUE THREAD

**Issue by kazuryyx** [dzikoysk/reposilite#1904]
### Request details

So, what if it was possible to add a javadocs viewer for example:

Lets say, having a repo up named like "SomeAPI" and you go to the page: "domain/javadoc/snapshots/me/somename/SomeAPI" It would give you a link that always points to the latest javadoc view where you can select all of the versions that are currently in "SomeAPI" and make you select one of the snapshot versions -> it opens the page on "domain/javadoc/snapshots/me/somename/SomeAPI/selectedVersion-SNAPSHOT" where people can then see the docs for this specific version, and not having to specify the version in a link.

**Comment by mfnalex** [comment]
I'd also like to see a "permanent" link that always shows the latest available javadocs - e.g. urltorepo/groupid/artifactid/javadoc-latest

**Comment by dzikoysk** [comment]
I think it was already request on our Discord server a while ago, so it's good to have an issue for that. Unfortunately, I'm slightly busy right now. 

Because javadocs viewer is a part of backend, not frontend, we'd need to generate such list server side with raw html, just like this one:

https://github.com/dzikoysk/reposilite/blob/62deca6d9a859a1991d81074b553109255f5fdd2/reposilite-backend/src/main/kotlin/com/reposilite/javadocs/JavadocView.kt#L31-L37

**Comment by TechnicJelle** [comment]
I was just looking for something like this, too.
And while I haven't found a way to see all the available javadocs in a nice list or dropdown, I did find out that there is a permanent link to the latest javadoc version, by putting "latest" instead of the version number in the URL.
Example:
`https://maven.reposilite.com/javadoc/releases/com/reposilite/reposilite/3.2.0`
↓
`https://maven.reposilite.com/javadoc/releases/com/reposilite/reposilite/latest`

Though, it might be nice if this was an actual redirect, instead of a different URL that has the same content on it.
Like how GitHub Releases does it: if click on this URL:
`https://github.com/dzikoysk/reposilite/releases/latest`
it redirects to
`https://github.com/dzikoysk/reposilite/releases/tag/3.5.14`

**Comment by kaanatesel** [comment]
I could obtain the list of available javadocs. However, as the index.html are create only once returned from the cache newly added javadocs will not be available for in the index.html that is already created.

One possible solution can be displaying can be available javadocs in different path. 
`<repository>/<group>/<artifact>/list`

exp : http:/blabla/javadoc/test-repo/org/slf4j/slf4j-simple/**list**

Other can be creating javadoc file each time the user request no caching. 

Which is more applicable to implement?

**Comment by kaanatesel** [comment]
Good idea,  any suggestion where I can listen deployments? and delete file?

**Comment by dzikoysk** [comment]
As far as I see, it's already there

https://github.com/dzikoysk/reposilite/blob/46ffce1fd65edbdea907e317df823242157b978e/reposilite-backend/src/main/kotlin/com/reposilite/javadocs/application/JavadocPlugin.kt#L54-L57

for deleting the file, just use regular file api, because cache is persisted on local fs.

**Comment by kaanatesel** [comment]
Could you re upload the javadoc file for the artifact it should invalidate index.html files cache.

**Comment by BlueTree242** [comment]
> Could you re upload the javadoc file for the artifact it should invalidate index.html files cache.

Sorry for not responding earlier. I just pushed a new version and the issue is still present.
